### PR TITLE
/tiers and /tags related fixes for Store REST API

### PR DESCRIPTION
**Pull request by malinthaprasan**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/gen/java/org/wso2/carbon/apimgt/rest/api/store/TiersApi.java [hunk 1]
```diff
@@ -42,10 +42,11 @@
     public Response tiersTierLevelGet(@ApiParam(value = "List API or Application type tiers.",required=true, allowableValues="{values=[api, application]}" ) @PathParam("tierLevel") String tierLevel,
     @ApiParam(value = "Maximum size of resource array to return.", defaultValue="25") @QueryParam("limit") Integer limit,
     @ApiParam(value = "Starting point within the complete list of items qualified.", defaultValue="0") @QueryParam("offset") Integer offset,
+    @ApiParam(value = "For cross-tenant invocations, this is used to specify the tenant domain, where the resource need to be \n  retirieved from."  )@HeaderParam("X-WSO2-Tenant") String xWSO2Tenant,
     @ApiParam(value = "Media types acceptable for the response. Default is JSON."  , defaultValue="JSON")@HeaderParam("Accept") String accept,
     @ApiParam(value = "Validator for conditional requests; based on the ETag of the formerly retrieved\nvariant of the resourec."  )@HeaderParam("If-None-Match") String ifNoneMatch)
     {
-    return delegate.tiersTierLevelGet(tierLevel,limit,offset,accept,ifNoneMatch);
+    return delegate.tiersTierLevelGet(tierLevel,limit,offset,xWSO2Tenant,accept,ifNoneMatch);
```
Comment (sinthuja) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/gen/java/org/wso2/carbon/apimgt/rest/api/store/TiersApi.java]: need to have spaces in between the parameters formatting issue.

Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/gen/java/org/wso2/carbon/apimgt/rest/api/store/TiersApi.java]: Please note this is part of a generated code using swagger-code-gen tool.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/TiersApiServiceImpl.java [hunk 2]
```diff
@@ -106,21 +115,29 @@ public Response tiersTierLevelGet(String tierLevel, Integer limit, Integer offse
...
+            if (!RestApiUtil.isTenantAvailable(requestedTenantDomain)) {
+                throw RestApiUtil.buildBadRequestException("Provided tenant domain '" + xWSO2Tenant + "' is invalid");
+            }
+
             TierDTO.TierLevelEnum tierType;
-            Tier foundTier;
-            APIConsumer apiConsumer = RestApiUtil.getLoggedInUserConsumer();
+            Tier foundTier = null;
             if (!StringUtils.isBlank(tierLevel)) {
                 if (TierDTO.TierLevelEnum.api.toString().equals(tierLevel)) {
```
Comment (lalaji) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/TiersApiServiceImpl.java]: Can a NPE appears here?

Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/TiersApiServiceImpl.java]: I think no.. as "TierDTO.TierLevelEnum.api.toString()" cannot become null as TierDTO.TierLevelEnum.api is a constant (enum)

Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/TiersApiServiceImpl.java]: This enum is named lowercase; please see https://github.com/wso2/carbon-apimgt/pull/1357#discussion_r47447320

**Pull request merged**